jbmc/src/java_bytecode/java_utils.cpp

```diff
@@ -436,3 +437,20 @@ const std::unordered_set<std::string> cprover_methods_to_ignore
   "endThread",
   "getCurrentThreadID"
 };
+
+/// \param type: type of new symbol
+/// \param basename_prefix: new symbol will be named
+///   function_name::basename_prefix$num
+/// \param source_location: new symbol source loc
+/// \param symbol_table: table to add the new symbol to
+/// \return fresh-named symbol with the requested name pattern
+symbolt &fresh_java_symbol(
+  const typet &type,
+  const std::string &basename_prefix,
+  const source_locationt &source_location,
+  symbol_table_baset &symbol_table)
+{
+  const std::string name_prefix = id2string(source_location.get_function());
```

Comment:
🚫  Unfortunately, this is in conflict with https://github.com/diffblue/cbmc/pull/2149
@tautschnig, we must come to a decision there.

Reply:
I strongly believe that the information in the `source_location` is a string aimed at the user, and should relate to whatever is found in the _source_. The actual identifier of the function (the key in `goto_functionst::function_map`) used internally may be entirely different. I don't know which one is required here.

Reply:
Here, something is required that is unique to a goto function. Reason: adding a function to a goto model must not affect the identifiers of another function. This was the source of many goto-diff bugs in the past.

Reply:
Then a `function_id` must be passed in. In some call sites in this PR this would be `symbol.name`, others require further digging.